apps/nowcasting-app/components/helpers/utils.test.ts

```diff
@@ -181,3 +182,59 @@ describe("prettyPrintDayLabelWithDate", () => {
     expect(result).toBe("Today");
   });
 });
+
+describe("check y-axis max value calculation", () => {
+  test("should return 0 when chart data is null", () => {
+    const result = utils.calculateChartYMax(null);
+    expect(result).toBe(0);
+  });
+
+  test("should return 0 when chart data is undefined", () => {
+    const result = utils.calculateChartYMax(undefined);
+    expect(result).toBe(0);
+  });
+
+  test("should return 0 when chart data is empty array", () => {
+    const result = utils.calculateChartYMax([]);
+    expect(result).toBe(0);
+  });
+
+  test("should round up to nearest 2000 with buffer of 1000", () => {
+    const chartData = [{ value: 12500 }];
+    const result = utils.calculateChartYMax(chartData);
+    expect(result).toBe(14000); // 12500 + 1000 = 13500 rounded up to nearest 2000, ie 14000
+  });
+
+  test("should handle multiple numeric values", () => {
+    const chartData = [
+      { value1: 5000, value2: 8000 },
```

Comment:
With the `ChartData` type in the function as per other comment below, this needs to fit that type too